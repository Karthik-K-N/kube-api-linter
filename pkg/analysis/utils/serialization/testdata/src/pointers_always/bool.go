package a

type TestBools struct {
	Bool bool `json:"bool"` // want "field Bool should have the omitempty tag." "field Bool should be a pointer."

	BoolWithOmitEmpty bool `json:"boolWithOmitEmpty,omitempty"` // want "field BoolWithOmitEmpty should be a pointer."

	BoolPtr *bool `json:"boolPtr"` // want "field BoolPtr should have the omitempty tag."

	BoolPtrWithOmitEmpty *bool `json:"boolPtrWithOmitEmpty,omitempty"`
}
