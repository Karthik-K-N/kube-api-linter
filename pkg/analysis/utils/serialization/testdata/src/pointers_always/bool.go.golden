package a

type TestBools struct {
	Bool *bool `json:"bool,omitempty"` // want "field Bool should have the omitempty tag." "field Bool should be a pointer."

	BoolWithOmitEmpty *bool `json:"boolWithOmitEmpty,omitempty"` // want "field BoolWithOmitEmpty should be a pointer."

	BoolPtr *bool `json:"boolPtr,omitempty"` // want "field BoolPtr should have the omitempty tag."

	BoolPtrWithOmitEmpty *bool `json:"boolPtrWithOmitEmpty,omitempty"`
}
